console/bin/console_server.js

```diff
@@ -15,136 +15,28 @@
 
-var express = require('express');
-var basic_auth = require('basic-auth');
-var path = require('path');
-var rhea = require('rhea');
-var WebSocketServer = require('ws').Server;
-var address_ctrl = require('../lib/address_ctrl.js').create();
+var address_ctrl = require('../lib/address_ctrl.js');
 var AddressList = require('../lib/address_list.js');
-var address_list = new AddressList();
-var auth_service = require('../lib/auth_service.js');
-var RouterStats = require('../lib/router_stats.js');
-var router_stats = new RouterStats();
-var BrokerStats = require('../lib/broker_stats.js');
-var broker_stats = new BrokerStats();
-var Registry = require('../lib/registry.js');
-var http = require('http');
-
-var app = express();
-app.get('/probe', function (req, res) { res.send('OK'); });
-
-var auth_ca_path = process.env.AUTH_SERVICE_CA || path.resolve('/opt/console/authservice-ca', 'tls.crt');
-
-var auth = function (req, res, next) {
-    var authrequired = function() {
-        res.set('WWW-Authenticate', 'Basic realm=Authorization Required');
-        return res.sendStatus(401);
-    };
-    var user = basic_auth(req);
-
-    auth_service.authenticate(user, auth_service.default_options(auth_ca_path)).then( next ).catch( authrequired );
-};
-
-app.use(auth);
-
-app.use('/', express.static(path.join(__dirname, '../www/')))
-
-var server = http.createServer(app);
-server.listen(8080);
-
-var ws_server = new WebSocketServer({server: server, 'path': '/websocket'});
-var amqp_container = rhea.create_container({autoaccept:false});
-ws_server.on('connection', function (ws) {
-    console.log('Accepted incoming websocket connection');
-    amqp_container.websocket_accept(ws);
-});
-
-var connections = new Registry();
-//connections.debug = true;
+var AddressSource = require('../lib/address_source.js');
+var ConsoleServer = require('../lib/console_server.js');
 
-var listeners = {};
-
-function subscribe(name, sender) {
-    listeners[name] = sender;
-    address_list.for_each(function (address) {
-        sender.send({subject:'address', body:address});
-    });
-    connections.for_each(function (conn) {
-        sender.send({subject:'connection', body:conn});
-    });
-    //TODO: poll for changes in address_types
-    address_ctrl.get_address_types().then(function (address_types) {
-        sender.send({subject:'address_types', body:address_types});
-    });
-}
-
-function unsubscribe(name) {
-    delete listeners[name];
+var address_list = new AddressList();
+var address_source = new AddressSource();
+address_source.on('addresses_defined', address_list.addresses_defined.bind(address_list));
+
+var console_server = new ConsoleServer(address_list, address_ctrl.create());
+console_server.listen();
+
+if (process.env.ADDRESS_SPACE_TYPE === 'broker') {
+    var BrokerController = require('../lib/broker_controller.js');
+    var bc = new BrokerController(console_server);
+    //TODO: have broker connect to controller? or have controller connect to broker?
```

Comment:
IMO have controller connect to broker seems to be the simplest since one does not have to add the outgoing connector + listener in the broker controller.

Reply:
Yes, I think I agree. The standard address space will be the other way round, but there is really no reason they should be the same.